src/webui/www/public/scripts/login.js

```diff
@@ -62,3 +115,19 @@ function submitLoginForm(event) {
     // clear the field
     passwordElement.value = '';
 }
+
+function tr(key) {
```

Comment:
Why do you introduce the `tr()` function? For familiarity with Qt?
The ` i18next.t()` is an overloaded function, and can return a default value like this:
```js
 i18next.t(key, key);
```

Docs: https://www.i18next.com/translation-function/essentials#passing-a-default-value

Reply:
Also there is the the following documentation page which I found from google but seems to have been dropped from the navigation menu of the official site (in the past I could navigate to it when I was also investigating i18next):
https://www.i18next.com/principles/fallback#key-fallback

Reply:
>Why do you introduce the tr() function? 
>The  i18next.t() is an overloaded function, and can return a default value like this:

I want it to return the key value when the mapped value is empty and it cannot do that.

Reply:
There is also a second reason related to i18next-parser which will emit a warning when extracting strings. But it is currently work arounded.

Reply:
> I want it to return the key value when the mapped value is empty (exists but empty, i.e. not yet translated) and `i18next.t()` cannot do that

Are you aware of the `returnEmptyString` config option? Link: https://www.i18next.com/overview/configuration-options#translation-defaults

Reply:
And I think setting `returnEmptyString = false` will make this `i18next.t(key)` also return the `key` if the it can't find a translated string.

Reply:
>Are you aware of the returnEmptyString config option? Link: https://www.i18next.com/overview/configuration-options#translation-defaults

I wasn't. Good finding!

>And I think setting returnEmptyString = false will make i18next.t(key) also return the key if it can't find a translated string.

True. PR updated.